packages/playout-gateway/src/atemUploader.ts

```diff
@@ -72,53 +88,67 @@ export class AtemUploadScript {
 			}
 		} else {
 			consoleLog('has no still')
-			throw Error('Atem appears to be missing still')
+			throw Error('Atem appears to be missing ' + type)
 		}
 	}
 
-	public async uploadToAtem(fileData: Buffer, stillIndex: number): Promise<void> {
-		const fileName = this.fileUrl.substr(-AtemMaxFilenameLength) // cannot be longer than 63 chars
-		if (!this.checkIfFileExistsOnAtem(fileName, stillIndex)) {
-			consoleLog('does not exist on ATEM')
+	public async uploadStillToAtem(fileName: string, fileData: Buffer, stillIndex: number): Promise<void> {
+		fileName = fileName.substr(-ATEM_MAX_FILENAME_LENGTH) // cannot be longer than 63 chars
+		if (!this.checkIfFileOrClipExistsOnAtem(fileName, stillIndex, AtemMediaPoolType.Still)) {
+			consoleLog(fileName + ' does not exist on ATEM')
 			await this.connection.clearMediaPoolStill(stillIndex)
 			await this.connection.uploadStill(stillIndex, fileData, fileName, '')
 		} else {
-			consoleLog('does exist on ATEM')
+			consoleLog(fileName + ' does exist on ATEM')
```

Comment:
Shouldn't the else branch throw an exception to indicate that the still has not been uploaded?

Reply:
That's not my intention, if the function successfully executes that indicates the media exists on the atem. If the media already exists on the atem no upload has to be done but that is not an error because the media will exist after the function returns. The whole functionality here is intended to be replaced by the package manager so I think it's justified to leave it as is

Reply:
If I put my super pedantic hat on I could say that you only know that an item with the same (truncated) name exists on the atem 🙃 
But if it's short lived code and we've never had any issues with this so far I guess it's fine to leave it like that until we replace this with the media manager.

Reply:
that is true, but why would we exception there?
If we are throwing an exception that implies that something is wrong, but nothing is wrong here. We are choosing to skip the upload as we believe the correct file is already there

Reply:
It might be that I don't know enough about the code that would call this function, but I guess my logic goes something like this:
- caller wants a file uploaded
- the function does not upload the file because there already is a file with the same name on the atem. Whether or not it is the same file is unknown
- the file that was requested uploaded is not uploaded, and that is an exception from the original expectation of the caller and should be flagged as such

But yeah, I might be missing information and/or constructing a problem where there is none :)